src/openagi/agent.py

```diff
@@ -435,20 +520,34 @@ def run(self, query: str, description: str,planned_tasks: Optional[List[Dict]] =
         self.memory.save_planned_tasks(tasks=list(task_lists.tasks.queue))
 
         if self.planner.autonomous:
-            return self.auto_workers_assignment(
+            result = self.auto_workers_assignment(
                 query=query, description=description, task_lists=task_lists
             )
         else:
             if self.workers:
-                return self.worker_task_execution(
+                result = self.worker_task_execution(
                     query=query,
                     description=description,
                     task_lists=task_lists,
                 )
             else:
-                return self.single_agent_execution(
+                result = self.single_agent_execution(
                     query=query, description=description, task_lists=task_lists
                 )
+        # Human feedback part
+        if self.memory.long_term:
+            if bad_feedback:
+                self.save_ltm("update", bad_session)
```

Comment:
add logging statement for the update in memory

Reply:
save_ltm is a function that handles both add and update, logging is incorporated in save_ltm, so not included here. #
```python
 def save_ltm(self, action_type: str, session: SessionDict):
        """
        Save a session to long-term memory by either adding or updating an existing session.

        :param action_type: Type of operation: 'add' or 'update'
        :param session: The SessionDict object containing session details
        """
        # Get feedback for plan and answer
        session.plan_feedback = self.input_action.execute(
            prompt=(
                f"Review the generated plan: \n{session.plan}\n"
                "If satisfied, press ENTER. \nOtherwise, describe the issue and suggest improvements:"
            )
        ).strip()

        session.ans_feedback = self.input_action.execute(
            prompt=(
                f"Review the generated answer: \n{session.answer}\n"
                "If satisfied, press ENTER. \nOtherwise, describe the issue and suggest improvements:"
            )
        ).strip()

        # Save or update based on the action_type
        if action_type == "add":
            self.memory.add_ltm(session)
            logging.info(f"Session added to long-term memory: {session}")
        elif action_type == "update":
            self.memory.update_ltm(session)
            logging.info(f"Session updated in long-term memory: {session}")
        else:
            raise ValueError("Invalid action_type. Use 'add' or 'update'.")

```